WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationService.java

```diff
@@ -0,0 +1,338 @@
+                        R.mipmap.app_icon))
+                .setAutoCancel(true)
+                .setContentIntent(PendingIntent.getActivity(SiteCreationService.this,
+                        AutoForeground.NOTIFICATION_ID_SUCCESS,
+                        getPendingIntent(),
+                        PendingIntent.FLAG_ONE_SHOT))
+                .build();
+    }
+
+    private Notification getFailureNotification(String content) {
+        return new NotificationCompat.Builder(this)
+                .setContentTitle(content)
+                .setSmallIcon(R.mipmap.app_icon)
+                .setLargeIcon(BitmapFactory.decodeResource(getApplicationContext().getResources(),
+                        R.mipmap.app_icon))
+                .setAutoCancel(true)
+                .setContentIntent(PendingIntent.getActivity(SiteCreationService.this,
+                        AutoForeground.NOTIFICATION_ID_FAILURE,
+                        getPendingIntent(),
+                        PendingIntent.FLAG_ONE_SHOT))
+                .build();
+    }
+
+    @Override
+    public int onStartCommand(@Nullable Intent intent, int flags, int startId) {
+        if (intent == null) {
+            return START_NOT_STICKY;
+        }
+
+        setState(SiteCreationPhase.NEW_SITE);
+
+        final String siteTitle = intent.getStringExtra(ARG_SITE_TITLE);
+        final String siteSlug = intent.getStringExtra(ARG_SITE_SLUG);
+        mSiteTagline = intent.getStringExtra(ARG_SITE_TAGLINE);
+        mSiteTheme = intent.getStringExtra(ARG_SITE_THEME);
+
+        final String language = LanguageUtils.getPatchedCurrentDeviceLanguage(this);
+
+        SiteStore.NewSitePayload newSitePayload =new SiteStore.NewSitePayload(
+                siteSlug,
+                siteTitle,
+                language,
+                SiteStore.SiteVisibility.PUBLIC,
+                false);
+        mDispatcher.dispatch(SiteActionBuilder.newCreateNewSiteAction(newSitePayload));
+        AppLog.i(T.NUX, "User tries to create a new site, title: " + siteTitle + ", SiteName: " + siteSlug);
+
+        return START_REDELIVER_INTENT;
+    }
+
+    private void activateTheme(final SiteModel site, final String themeId) {
+        WordPress.getRestClientUtils().setTheme(site.getSiteId(), themeId, new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject response) {
+                ThemeTable.setCurrentTheme(WordPress.wpDB.getDatabase(), String.valueOf(site.getSiteId()), themeId);
+
+                setState(SiteCreationPhase.SUCCESS);
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                setState(SiteCreationPhase.FAILURE);
+//                ToastUtils.showToast(ThemeBrowserActivity.this, R.string.theme_activation_error,
+//                        ToastUtils.Duration.SHORT);
```

Comment:
let's not forget to remove this commented code here before :shipit: 🎗

Reply:
Truth be told, I left those in to handle right after we had a "green light" for the AutoForeground mechanism. Technically this PR is not ready to be merged yet. But thanks for pointing it out! I will remove the comments so we can ideally merge the PR. 👍

Reply:
Removed with 311a2cd.